lib/vagrant-openshift/helper/command_helper.rb

```diff
@@ -160,7 +160,9 @@ def repo_checkout_bash_command(repo, url)
         git fetch origin
         git fetch --prune origin +refs/tags/*:refs/tags/*
         git reset --merge
-        git checkout #{branch}
+        if ! git checkout #{branch}; then
+          git checkout -b #{branch} origin/#{branch}
```

Comment:
@danmcp, `git checkout -b #{branch} origin/#{branch}` won't work if a local branch already exists.  eg, `git checkout -b master origin/master` will fail for most repos that have been recently cloned.